app/src/lib/helpers/repo-rules.ts

```diff
@@ -64,11 +65,13 @@ export function useRepoRulesLogic(
  * Parses the GitHub API response for a branch's repo rules into a more useable
  * format.
  */
-export function parseRepoRules(
+export async function parseRepoRules(
   rules: ReadonlyArray<IAPIRepoRule>,
-  rulesets: ReadonlyMap<number, IAPIRepoRuleset>
-): RepoRulesInfo {
+  rulesets: ReadonlyMap<number, IAPIRepoRuleset>,
+  repository: Repository
+): Promise<RepoRulesInfo> {
   const info = new RepoRulesInfo()
+  let gitConfigKeySet: boolean | null = null
```

Comment:
```suggestion
  let gitConfigKeySet: boolean | null = null
```

It doesn't seem like this is needed to be defined before hand, and should be able to in the switch statement replace the first `if` with `const gitConfigKeySet = ...`

Reply:
I declared it up there to cache it across rule checks--if there are multiple rules that require signed commits, there's no need to query the git config more than once because it's always the same repo. Tiny optimization but I figured it made sense.

Reply:
That makes sense. However, I think since currently it is only one, we would prefer to keep in with our engineering value about maintaining [Immutability](https://github.com/desktop/desktop/blob/development/docs/contributing/engineering-values.md#immutability-is-good). 

In the future if it is needed, we can likely do something like memoize for this sort of optimization. :)

Reply:
Sorry I didn't explain properly. Any number of rulesets can apply to the user's branch, which can result in having multiple of this type of rule. That's why I put this in here--this case may be hit multiple times right now.

Whether the underlying git operation is slow enough to justify this optimization against the code style is another question, but it is at least a current optimization, it's not future-proofing.